nservicebus/messaging/uniformsession.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Uniform Session between message session and pipeline
+summary: Uniform Session allows to bridge the gap between the message session and the pipeline context and enables dependency injection.
+reviewed: 2017-10-27
+component: UniformSession
+---
+
+NServiceBus v6 and higher introduced a few design changes that how code that is using previous versions is structured. The [changes](/nservicebus/upgrades/5to6/moving-away-from-ibus.md) are:
+
+- In previous versions, the `IBus` interface was automatically registered in the IOC container. In Version 6, the new context-aware interfaces, for example, `IEndpointInstance`, `IMessageSession` and `IMessageHandlerContext`, etc., are not automatically registered in dependency injection.
+- In Versions 5 and below, when a custom component was registered in the container, the custom component had access to the `IBus` instance via dependency injection.
+- Clear separation of concerns between message operations outside the message handling pipeline (`IEndpointInstance` and `IMessageSession`) vs. inside the message handling pipeline (`IMessageHandlerContext`). Operations inside the message handling pipeline enlist in the transaction available and participate in the batching operations of the transport. Operations outside the pipeline don't have those capabilities. The duality of the `IBus` interface was removed by introducing separate explicit interfaces for each usage scenario.
+- Message operations are asynchronous by default
+
+These changes lead to better designed and safer user code because:
+
+- Child containers no longer need to be mutated when a message arrives to rebind the current message handler context which simplifies container integration especially for immutable containers
```

Comment:
I'm not sure we want to justify these decision in the documentation of the plugin. Maybe just keep the first sentence (`These changes lead to better designed and safer user code` ) to the `NServiceBus v6 and higher introduced a few design changes[...]` paragraph above?

Reply:
IMHO I think it is good to have the justifications somewhere. We could also add them to the upgrade guide?

Reply:
I think the upgrade guide would suit better